File: 442-find-all-duplicates-in-an-array/442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> v;
        int i=0;
        while(i<nums.size())
        {
            int cor=nums[i]-1;
            if(nums[i]!=nums[cor])
            {
                int temp=nums[i];
                nums[i]=nums[cor];
                nums[cor]=temp;
            }
            else 
                i++;
        }
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i]!=i+1)
            {
                v.push_back(nums[i]);
                
            }
        }
        return v;
        
    }
};

File: 442-find-all-duplicates-in-an-array/442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] a) {
        int i=0;
        while(i<a.length)
        {
            int cor=a[i]-1;
            if(a[i]!=a[cor])
            {
                int temp=a[i];
                a[i]=a[cor];
                a[cor]=temp;
            }
            else
            {
                i++;}
        }
        List<Integer> l=new ArrayList<Integer>();
        for(int x=0;x<a.length;x++)
        {
            if(a[x]!=x+1)
            {
                l.add(a[x]);
            }
        }
        return l;
        
    }
}